com.teamscale.polarion.plugin/src/com/teamscale/polarion/plugin/model/LinkBundle.java

```diff
@@ -0,0 +1,41 @@
+package com.teamscale.polarion.plugin.model;
+
+/**
+ * This is a helper object that maintains link data and is used to process and generate opposite
+ * link changes in the plugin. *
+ */
+public class LinkBundle {
```

Comment:
Personally I am more of a fan of immutable objects (in combination with builders if required).
Do you think it would be feasible to adapt the models accordingly?

Reply:
I made LinkBundle immutable (attributes as final, no set methods) and the same for a few other model classes. 

However, WorkItemChange, WorkItemFieldDiff, and WorkItemForJson are not feasible as the logic goes back to those already created objects and changes their state when there are link changes (as we need to create the opposite link of the link changes 'manually' as Polarion doesn't give us that 'out of the box'). 

It's technically possible to turn these others immutable  but it would require a major refactoring and would probably hurt performance to some degree (object re-creation rather than changing the state of an existing one), which I'd like to avoid now.